contracts/sfc/NodeDriver.sol

```diff
@@ -1,230 +1,19 @@
-        driver.updateNetworkRules(diff);
-    }
-
-    function updateMinGasPrice(uint256 minGasPrice) external onlySFC {
-        // prettier-ignore
-        driver.updateNetworkRules(bytes(strConcat("{\"Economy\":{\"MinGasPrice\":", uint256ToStr(minGasPrice), "}}")));
-    }
-
-    function updateNetworkVersion(uint256 version) external onlyOwner {
-        driver.updateNetworkVersion(version);
-    }
-
-    function advanceEpochs(uint256 num) external onlyOwner {
-        driver.advanceEpochs(num);
-    }
-
-    function updateValidatorWeight(uint256 validatorID, uint256 value) external onlySFC {
-        driver.updateValidatorWeight(validatorID, value);
-    }
-
-    function updateValidatorPubkey(uint256 validatorID, bytes calldata pubkey) external onlySFC {
-        driver.updateValidatorPubkey(validatorID, pubkey);
-    }
-
-    function setGenesisValidator(
-        address _auth,
-        uint256 validatorID,
-        bytes calldata pubkey,
-        uint256 status,
-        uint256 createdEpoch,
-        uint256 createdTime,
-        uint256 deactivatedEpoch,
-        uint256 deactivatedTime
-    ) external onlyDriver {
-        sfc.setGenesisValidator(
-            _auth,
-            validatorID,
-            pubkey,
-            status,
-            createdEpoch,
-            createdTime,
-            deactivatedEpoch,
-            deactivatedTime
-        );
-    }
-
-    function setGenesisDelegation(
-        address delegator,
-        uint256 toValidatorID,
-        uint256 stake,
-        uint256 lockedStake,
-        uint256 lockupFromEpoch,
-        uint256 lockupEndTime,
-        uint256 lockupDuration,
-        uint256 earlyUnlockPenalty,
-        uint256 rewards
-    ) external onlyDriver {
-        sfc.setGenesisDelegation(
-            delegator,
-            toValidatorID,
-            stake,
-            lockedStake,
-            lockupFromEpoch,
-            lockupEndTime,
-            lockupDuration,
-            earlyUnlockPenalty,
-            rewards
-        );
-    }
-
-    function deactivateValidator(uint256 validatorID, uint256 status) external onlyDriver {
-        sfc.deactivateValidator(validatorID, status);
-    }
-
-    function sealEpochValidators(uint256[] calldata nextValidatorIDs) external onlyDriver {
-        sfc.sealEpochValidators(nextValidatorIDs);
-    }
-
-    function sealEpoch(
-        uint256[] calldata offlineTimes,
-        uint256[] calldata offlineBlocks,
-        uint256[] calldata uptimes,
-        uint256[] calldata originatedTxsFee,
-        uint256 usedGas
-    ) external onlyDriver {
-        sfc.sealEpoch(offlineTimes, offlineBlocks, uptimes, originatedTxsFee, usedGas);
-    }
-
-    function isContract(address account) internal view returns (bool) {
-        uint256 size;
-        // solhint-disable-next-line no-inline-assembly
-        assembly {
-            size := extcodesize(account)
-        }
-        return size > 0;
-    }
+interface EVMWriter {
```

Comment:
it seems it would be better to move EVMWriter interface into a standalone file...